diff --git a/components/spi_flash/cache_utils.c b/components/spi_flash/cache_utils.c
--- a/components/spi_flash/cache_utils.c
+++ b/components/spi_flash/cache_utils.c
@@ -60,6 +60,9 @@ static inline bool esp_task_stack_is_sane_cache_disabled(void)
     return esp_ptr_in_dram(sp)
 #if CONFIG_ESP_SYSTEM_ALLOW_RTC_FAST_MEM_AS_HEAP
            || esp_ptr_in_rtc_dram_fast(sp)
 #endif
+#if SOC_MEM_SPM_SUPPORTED
+           || esp_ptr_in_spm(sp)
+#endif
            ;
 }
